src/ragas/integrations/llama_index.py

```diff
@@ -82,8 +83,15 @@ def evaluate(
     retrieved_contexts: t.List[t.List[str]] = []
     results = exec.results()
     for r in results:
-        responses.append(r.response)
-        retrieved_contexts.append([n.node.text for n in r.source_nodes])
+        # Handle failed jobs which are recorded as NaN in the executor
+        if isinstance(r, float) and math.isnan(r):
+            responses.append("")
+            retrieved_contexts.append([])
+        else:
+            # Cast to LlamaIndex Response type for proper type checking
+            response = t.cast("LlamaIndexResponse", r)
```

Comment:
This'll be hard on type hints.

Probably better to take `from llama_index.core.base.response.schema import Response as LlamaIndexResponse`